libqtile/scripts/launch.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2025, elParaguayo
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the \"Software\"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+Command-line wrapper to launch process from within qtile
+"""
+
+import argparse
+import os
+import socket
+
+
+def launch(opts) -> None:
+    fd = opts.fd
+
+    if fd > 0:
+        # Create a socket from the file descriptor
+        sock = socket.fromfd(fd, socket.AF_INET, socket.SOCK_STREAM)
+
+        # Tell qtile we're ready to listen for their message
+        sock.sendall(b"READY")
+
+        # Receive OK message (but we're not checking the contents, we just
+        # block until we receive the data)
+        _ = sock.recv(2)
```

Comment:
I would print out what you get back here. The fact that it's not working suggests something strange is happening.

Reply:
Where are we printing it? stdout and stderr have been redirected to `/dev/null`. I suppose we could not do that redirect until just before we `exec`.